### ISSUE TITLE
Fatal error: unreachable code in ` , line 0`

### ISSUE THREAD

**Issue by derekzTu** [v8-riscv/v8#85]
**Describe the bug**
Running `test/intl/date-format/format-range.js` triggers an internal error

**To Reproduce**
1. Build release build (see instructions [here](https://github.com/v8-riscv/v8/wiki/simulator-build))
2. Run program:
```bash
out/riscv64.release/d8 --test test/intl/assert.js test/intl/utils.js test/intl/date-format/format-range.js --random-seed=-819243071 --stress-opt --always-opt --no-liftoff --stress-lazy-source-positions --nohard-abort --testing-d8-test-runner --allow-natives-syntax
```
3. See error
```bash
=== intl/date-format/format-range ===
============ Stress 1/5 ============
============ Stress 2/5 ============
============ Stress 3/5 ============
============ Stress 4/5 ============
============ Stress 5/5 ============
#
# Fatal error in , line 0
# unreachable code
#
#
#
#FailureMessage Object: 0x7ffc90639100
```

**Expected behavior**
No error messages
**Errors/Logs**
Full stack trace:
```bash
==== C stack trace ===============================

    /root/home/v8_root/v8/out/riscv64.release/d8(v8::base::debug::StackTrace::StackTrace()+0x13) [0x5580bd6f75b3]
    /root/home/v8_root/v8/out/riscv64.release/d8(+0x12bcbad) [0x5580bd6f6bad]
    /root/home/v8_root/v8/out/riscv64.release/d8(V8_Fatal(char const*, ...)+0x150) [0x5580bd6f1290]
    /root/home/v8_root/v8/out/riscv64.release/d8(v8::internal::SafepointTable::FindEntry(unsigned long) const+0xc4) [0x5580bcc30d54]
    /root/home/v8_root/v8/out/riscv64.release/d8(v8::internal::Code::GetSafepointEntry(unsigned long)+0x50) [0x5580bce1bb10]
    /root/home/v8_root/v8/out/riscv64.release/d8(v8::internal::OptimizedFrame::GetDeoptimizationData(int*) const+0xcb) [0x5580bcc93ffb]
    /root/home/v8_root/v8/out/riscv64.release/d8(v8::internal::OptimizedFrame::Summarize(std::__1::vector<v8::internal::FrameSummary, std::__1::allocator<v8::internal::FrameSummary> >*) const+0x76) [0x5580bcc93926]
    /root/home/v8_root/v8/out/riscv64.release/d8(+0x86003a) [0x5580bcc9a03a]
    /root/home/v8_root/v8/out/riscv64.release/d8(v8::internal::Isolate::CaptureAndSetSimpleStackTrace(v8::internal::Handle<v8::internal::JSReceiver>, v8::internal::FrameSkipMode, v8::internal::Handle<v8::internal::Object>)+0x62) [0x5580bcc9a7a2]
    /root/home/v8_root/v8/out/riscv64.release/d8(v8::internal::ErrorUtils::Construct(v8::internal::Isolate*, v8::internal::Handle<v8::internal::JSFunction>, v8::internal::Handle<v8::internal::Object>, v8::internal::Handle<v8::internal::Object>, v8::internal::FrameSkipM$
de, v8::internal::Handle<v8::internal::Object>, v8::internal::ErrorUtils::StackTraceCollection)+0xca) [0x5580bccad26a]
    /root/home/v8_root/v8/out/riscv64.release/d8(v8::internal::ErrorUtils::MakeGenericError(v8::internal::Isolate*, v8::internal::Handle<v8::internal::JSFunction>, v8::internal::MessageTemplate, v8::internal::Handle<v8::internal::Object>, v8::internal::Handle<v8::inter$
al::Object>, v8::internal::Handle<v8::internal::Object>, v8::internal::FrameSkipMode)+0x166) [0x5580bccad536]
    /root/home/v8_root/v8/out/riscv64.release/d8(v8::internal::Factory::NewError(v8::internal::Handle<v8::internal::JSFunction>, v8::internal::MessageTemplate, v8::internal::Handle<v8::internal::Object>, v8::internal::Handle<v8::internal::Object>, v8::internal:[10/1790]
8::internal::Object>)+0x4f) [0x5580bccd045f]
    /root/home/v8_root/v8/out/riscv64.release/d8(v8::internal::Object v8::internal::DateTimeFormatRange<v8::internal::JSArray>(v8::internal::BuiltinArguments, v8::internal::Isolate*, char const*, v8::internal::MaybeHandle<v8::internal::JSArray> (*)(v8::internal::Isolate
*, v8::internal::Handle<v8::internal::JSDateTimeFormat>, double, double))+0x185) [0x5580bcbf9e85]
    /root/home/v8_root/v8/out/riscv64.release/d8(v8::internal::Builtin_DateTimeFormatPrototypeFormatRange(int, unsigned long*, v8::internal::Isolate*)+0x4d) [0x5580bcbeae0d]
    /root/home/v8_root/v8/out/riscv64.release/d8(v8::internal::Simulator::SoftwareInterrupt()+0x4c3) [0x5580bd1b6f83]
    /root/home/v8_root/v8/out/riscv64.release/d8(v8::internal::Simulator::InstructionDecode(v8::internal::Instruction*)+0x1d0) [0x5580bd1b57a0]
    /root/home/v8_root/v8/out/riscv64.release/d8(v8::internal::Simulator::Execute()+0xab) [0x5580bd1be93b]
    /root/home/v8_root/v8/out/riscv64.release/d8(v8::internal::Simulator::CallInternal(unsigned long)+0xab) [0x5580bd1bea4b]
    /root/home/v8_root/v8/out/riscv64.release/d8(v8::internal::Simulator::CallImpl(unsigned long, int, long const*)+0x294) [0x5580bd1bee44]
    /root/home/v8_root/v8/out/riscv64.release/d8(+0x8538ab) [0x5580bcc8d8ab]
    /root/home/v8_root/v8/out/riscv64.release/d8(v8::internal::Execution::Call(v8::internal::Isolate*, v8::internal::Handle<v8::internal::Object>, v8::internal::Handle<v8::internal::Object>, int, v8::internal::Handle<v8::internal::Object>*)+0xf6) [0x5580bcc8cbf6]
    /root/home/v8_root/v8/out/riscv64.release/d8(v8::Script::Run(v8::Local<v8::Context>)+0x273) [0x5580bcb5afc3]
    /root/home/v8_root/v8/out/riscv64.release/d8(v8::Shell::ExecuteString(v8::Isolate*, v8::Local<v8::String>, v8::Local<v8::Value>, v8::Shell::PrintResult, v8::Shell::ReportExceptions, v8::Shell::ProcessMessageQueue)+0x5f9) [0x5580bcb34299]
    /root/home/v8_root/v8/out/riscv64.release/d8(v8::SourceGroup::Execute(v8::Isolate*)+0x24a) [0x5580bcb3ee9a]
    /root/home/v8_root/v8/out/riscv64.release/d8(v8::Shell::RunMain(v8::Isolate*, int, char**, bool)+0x257) [0x5580bcb418c7]
    /root/home/v8_root/v8/out/riscv64.release/d8(v8::Shell::Main(int, char**)+0xd19) [0x5580bcb43299]
    /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xe7) [0x7fc61558ab97]
    /root/home/v8_root/v8/out/riscv64.release/d8(_start+0x2a) [0x5580bcb2f7aa]
```

**Additional context**
There are additional tests that also have this error

Test:
`test-api/InternalFieldsSubclassing`

Command:
```bash
out/riscv64.release/cctest test-api/InternalFieldsSubclassing --random-seed=1724000608 --stress-opt --always-opt --no-liftoff --stress-lazy-source-positions --nohard-abort --testing-d8-test-runner
```
-------

Test:
`test/message/wasm-finish-compilation.js`

Command:
```bash
out/riscv64.release/d8 --test test/message/wasm-finish-compilation.js --random-seed=-1663505519 --stress-opt --always-opt --no-liftoff --stress-lazy-source-positions --nohard-abort --testing-d8-test-runner --no-stress-opt
```
-------

Test:
`test/inspector/debugger/max-async-call-chain-depth.js`

Command:
```bash
out/riscv64.release/inspector-test test/inspector/protocol-test.js test/inspector/debugger/max-async-call-chain-depth.js --random-seed=-345118250 --stress-opt --always-opt --no-liftoff --stress-lazy-source-positions --nohard-abort --testing-d8-test-runner
```
-------

Test:
`test/debugger/debug/debug-enable-disable-breakpoints.js`

Command:
```bash
 out/riscv64.release/d8 --test test/mjsunit/mjsunit.js test/debugger/test-api.js test/debugger/debug/debug-enable-disable-breakpoints.js --random-seed=2018349753 --stress-opt --always-opt --no-liftoff --stress-lazy-source-positions --nohard-abort --testing-d8-test-runner --enable-inspector --allow-natives-syntax
```
-------

Test:
`test/debugger/debug/debug-receiver.js`

Command:
```bash
out/riscv64.release/d8 --test test/mjsunit/mjsunit.js test/debugger/test-api.js test/debugger/debug/debug-receiver.js --random-seed=2018349753 --stress-opt --always-opt --no-liftoff --stress-lazy-source-positions --nohard-abort --testing-d8-test-runner --enable-inspector --allow-natives-syntax
```
-------

Test:
`test/debugger/debug/debug-step-3.js`

Command:
```bash
out/riscv64.release/d8 --test test/mjsunit/mjsunit.js test/debugger/test-api.js test/debugger/debug/debug-step-3.js --random-seed=2018349753 --stress-opt --always-opt --no-liftoff --stress-lazy-source-positions --nohard-abort --testing-d8-test-runner --enable-inspector --allow-natives-syntax
```
-------

Test:
`test/debugger/debug/es8/async-debug-builtin-predictions.js`

Command:
```bash
out/riscv64.release/d8 --test test/mjsunit/mjsunit.js test/debugger/test-api.js test/debugger/debug/es8/async-debug-builtin-predictions.js --random-seed=2018349753 --stress-opt --always-opt --no-liftoff --stress-lazy-source-positions --nohard-abort --testing-d8-test-runner --enable-inspector --allow-natives-syntax
```
-------
Test:
`test/debugger/debug/harmony/modules-debug-scopes1.mjs`

Command:
```bash
out/riscv64.release/d8 --test test/mjsunit/mjsunit.js test/debugger/test-api.js test/debugger/debug/harmony/modules-debug-scopes1.mjs --random-seed=2018349753 --stress-opt --always-opt --no-liftoff --stress-lazy-source-positions --nohard-abort --testing-d8-test-runner --allow-natives-syntax --noanalyze-environment-liveness --enable-inspector --allow-natives-syntax
```
-------
Test:
`test/debugger/debug/side-effect/debug-evaluate-no-side-effect-builtins.js`

Command:
```bash
out/riscv64.release/d8 --test test/mjsunit/mjsunit.js test/debugger/test-api.js test/debugger/debug/side-effect/debug-evaluate-no-side-effect-builtins.js --random-seed=2018349753 --stress-opt --always-opt --no-liftoff --stress-lazy-source-positions --nohard-abort --testing-d8-test-runner --no-enable-one-shot-optimization --enable-inspector --allow-natives-syntax
```
-------

Test:
` test/mjsunit/wasm/grow-memory.js`

Command:
```bash
out/riscv64.release/d8 --test test/mjsunit/mjsunit.js test/mjsunit/wasm/grow-memory.js --random-seed=-534112441 --stress-opt --always-opt --no-liftoff --stress-lazy-source-positions --nohard-abort --testing-d8-test-runner --expose-wasm --stress-compaction
```
-------

Test:
`test/mjsunit/array-push8.js`

Command:
```bash
out/riscv64.release/d8 --test test/mjsunit/mjsunit.js test/mjsunit/array-push8.js --random-seed=-534112441 --stress-opt --always-opt --no-liftoff --stress-lazy-source-positions --nohard-abort --testing-d8-test-runner --allow-natives-syntax
```
-------

Test:
`test/mjsunit/messages.js`

Command:
```bash
out/riscv64.release/d8 --test test/mjsunit/mjsunit.js test/mjsunit/messages.js --random-seed=-534112441 --stress-opt --always-opt --no-liftoff --stress-lazy-source-positions --nohard-abort --testing-d8-test-runner --allow-natives-syntax --stack-size=100 --harmony
```
-------

Test:
`test/mjsunit/es6/default-parameters.js`

Command:
```bash
out/riscv64.release/d8 --test test/mjsunit/mjsunit.js test/mjsunit/es6/default-parameters.js --random-seed=-534112441 --stress-opt --always-opt --no-liftoff --stress-lazy-source-positions --nohard-abort --testing-d8-test-runner
```
-------

Test:
`test/mjsunit/es6/reflect-apply.js`

Command:
```bash
out/riscv64.release/d8 --test test/mjsunit/mjsunit.js test/mjsunit/es6/reflect-apply.js --random-seed=-534112441 --stress-opt --always-opt --no-liftoff --stress-lazy-source-positions --nohard-abort --testing-d8-test-runner
```

-------

Test:
`test/mjsunit/es6/reflect-construct.js`

Command:
```bash
out/riscv64.release/d8 --test test/mjsunit/mjsunit.js test/mjsunit/es6/reflect-construct.js --random-seed=-534112441 --stress-opt --always-opt --no-liftoff --stress-lazy-source-positions --nohard-abort --testing-d8-test-runner --allow-unsafe-function-constructor
```
------

Test:
` test/mjsunit/es6/super.js`

Command:
```bash
 out/riscv64.release/d8 --test test/mjsunit/mjsunit.js test/mjsunit/es6/super.js --random-seed=-534112441 --stress-opt --always-opt --no-liftoff --stress-lazy-source-positions --nohard-abort --testing-d8-test-runner --allow-natives-syntax
```
-------

Test:
`test/mjsunit/es6/typedarray.js`

Command:
```bash
out/riscv64.release/d8 --test test/mjsunit/mjsunit.js test/mjsunit/es6/typedarray.js --random-seed=-534112441 --stress-opt --always-opt --no-liftoff --stress-lazy-source-positions --nohard-abort --testing-d8-test-runner --allow-natives-syntax
```
-------
Test:
`test/mjsunit/harmony/atomics.js`

Command:
```bash
out/riscv64.release/d8 --test test/mjsunit/mjsunit.js test/mjsunit/harmony/atomics.js --random-seed=-534112441 --stress-opt --always-opt --no-liftoff --stress-lazy-source-positions --nohard-abort --testing-d8-test-runner --harmony-sharedarraybuffer
```
-------
Test:
`test/mjsunit/regress/regress-145201.js`

Command:
```bash
out/riscv64.release/d8 --test test/mjsunit/mjsunit.js test/mjsunit/regress/regress-145201.js --random-seed=-534112441 --stress-opt --always-opt --no-liftoff --stress-lazy-source-positions --nohard-abort --testing-d8-test-runner
```
-------
Test:
`test/mjsunit/regress/regress-3926.js`

Command:
```bash
out/riscv64.release/d8 --test test/mjsunit/mjsunit.js test/mjsunit/regress/regress-3926.js --random-seed=-534112441 --stress-opt --always-opt --no-liftoff --stress-lazy-source-positions --nohard-abort --testing-d8-test-runner
```
-------
Test:
`test/mjsunit/regress/regress-4815.js`

Command:
```bash
 out/riscv64.release/d8 --test test/mjsunit/mjsunit.js test/mjsunit/regress/regress-4815.js --random-seed=-534112441 --stress-opt --always-opt --no-liftoff --stress-lazy-source-positions --nohard-abort --testing-d8-test-runner
```
-------
Test:
`test/mjsunit/regress/regress-crbug-706642.js`

Command:
```bash
out/riscv64.release/d8 --test test/mjsunit/mjsunit.js test/mjsunit/regress/regress-crbug-706642.js --random-seed=-534112441 --stress-opt --always-opt --no-liftoff --stress-lazy-source-positions --nohard-abort --testing-d8-test-runner --allow-natives-syntax
```
-------
Test:
`test/mjsunit/regress/regress-crbug-981701.js `

Command:
```bash
out/riscv64.release/d8 --test test/mjsunit/mjsunit.js test/mjsunit/regress/regress-crbug-981701.js --random-seed=-534112441 --stress-opt --always-opt --no-liftoff --stress-lazy-source-positions --nohard-abort --testing-d8-test-runner --throws --enable-lazy-source-positions --stress-lazy-source-positions
```
-------
Test:
`test/mjsunit/regress/wasm/regress-02862.js`
Command:
```bash
out/riscv64.release/d8 --test test/mjsunit/mjsunit.js test/mjsunit/regress/wasm/regress-02862.js --random-seed=-534112441 --stress-opt --always-opt --no-liftoff --stress-lazy-source-positions --nohard-abort --testing-d8-test-runner --random-seed=1557792826 --expose-gc --invoke-weak-callbacks --omit-quit --gc-interval=469 --validate-asm
```
-------
Test:
`test/mjsunit/regress/wasm/regress-02862.js`

Command:
```bash
out/riscv64.release/d8 --test test/mjsunit/mjsunit.js test/mjsunit/regress/wasm/regress-02862.js --random-seed=-534112441 --stress-opt --always-opt --no-liftoff --stress-lazy-source-positions --nohard-abort --testing-d8-test-runner --random-seed=1557792826 --expose-gc --invoke-weak-callbacks --omit-quit --gc-interval=469 --validate-asm
```
-------
Test:
`test/mjsunit/wasm/atomics.js`

Command:
```bash
out/riscv64.release/d8 --test test/mjsunit/mjsunit.js test/mjsunit/wasm/atomics.js --random-seed=-534112441 --stress-opt --always-opt --no-liftoff --stress-lazy-source-positions --nohard-abort --testing-d8-test-runner --experimental-wasm-threads
```
-------
Test:
`test/mjsunit/wasm/globals.js`

Command:
```bash
out/riscv64.release/d8 --test test/mjsunit/mjsunit.js test/mjsunit/wasm/globals.js --random-seed=-534112441 --stress-opt --always-opt --no-liftoff --stress-lazy-source-positions --nohard-abort --testing-d8-test-runner --expose-wasm
```
-------
Test:
`test/mjsunit/wasm/mutable-globals.js`

Command:
```bash
out/riscv64.release/d8 --test test/mjsunit/mjsunit.js test/mjsunit/wasm/mutable-globals.js --random-seed=-534112441 --stress-opt --always-opt --no-liftoff --stress-lazy-source-positions --nohard-abort --testing-d8-test-runner
```
-------
Test:
`test/mjsunit/wasm/unicode-validation.js`

Command:
```bash
out/riscv64.release/d8 --test test/mjsunit/mjsunit.js test/mjsunit/wasm/unicode-validation.js --random-seed=-534112441 --stress-opt --always-opt --no-liftoff --stress-lazy-source-positions --nohard-abort --testing-d8-test-runner --expose-wasm
```

**Comment by lazyparser** [comment]
Is anyone working on this bug?

**Comment by thomasyonug** [comment]
> Is anyone working on this bug?

I'll take a look.

**Comment by thomasyonug** [comment]
It's fixed by pr #139
